src/gazebo_mavlink_interface.cpp

```diff
@@ -411,8 +411,8 @@ void GazeboMavlinkInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf
 
   if (_sdf->HasElement("imu_rate")) {
     imu_rate_ = _sdf->GetElement("imu_rate")->Get<int>();
-    imu_rate_ = 1/imu_rate_;
   }
+  imu_rate_ = 1/imu_rate_;
```

Comment:
This is not correct if the above if clause is not executed. You're inverting a member variable which was set to 250 now to 1/250 = 0.004, which is not a rate but an interval.

Don't reuse variables like you do above. It creates programming errors like the one here. Use two different variables, one for the setting and one for the IMU rate. Name them consistently and choose concise naming (is it a rate or an interval?)

Reply:
The compiler will fill the CPU registers efficiently. Using two variables doesn't mean its more instructions - it's the compilers job to do this, its our job to keep the software readable and maintainable.

Reply:
I added a variable called imu_update_interval_ that replaces imu_rate_ in this line.